asposeimagingcloud/api/imaging_api.py

```diff
@@ -27,27 +27,83 @@
 from __future__ import absolute_import
 
 import six
-from asposeimagingcloud.rest import ApiException
+
 from asposeimagingcloud.api_client import ApiClient
+from asposeimagingcloud.configuration import Configuration
+from asposeimagingcloud.rest import ApiException
 
 
 class ImagingApi(object):
     """
     Aspose.Imaging Cloud API
 
-    :param app_key: The app key.
-    :param app_sid: The app sid.
-    :param base_url: The base URL.
-    :param api_version: API version.
     """
-    def __init__(self, app_key, app_sid, base_url=None, api_version=None):
-        self.api_client = ApiClient()
-        self.api_client.configuration.api_key['api_key'] = app_key
-        self.api_client.configuration.api_key['app_sid'] = app_sid
-        if base_url:
-            self.api_client.configuration.host = base_url
-        if api_version:
-            self.api_client.configuration.api_version = api_version
+
+    def __init__(self, app_key=None, app_sid=None, base_url=None,
```

Comment:
Python does not support constructor overloads, so I've extended ImagingApi constructor with new optional parameters, making it compatible with previous versions. Also I've added two factory methods `ImagingApi.create_cloud` and `ImagingApi.create_on_premise` with more strict parameters to improve usability.

Reply:
@s-zubov would you mind to remove these factory methods (at least for now)? Still, you are free to add a task with a proposition to unify all SDKs by adding such methods and assign it to me - I'll think about it a bit later and if approved, I'll re-assign it to you. Thanks.

Reply:
@MaxCrank I've removed these methods, now there is only one constructor with multiple optional parameters. `on_premise` parameter is not set explicitly and is determined by input data (`app_sid`, `app_key` and `base_url`).